### ISSUE TITLE
MacOs - Linux based - Possibility?

### ISSUE THREAD

**Issue by robot198** [jakeday/linux-surface#503]
So, I know this is **INCREDIBLY** far-fetched. Most likely not even possible. But i was wondering, if Mac Os X is linux-based than shouldn't there be a way to port this to Mac?
I have a surface laptop, it's been done on surface pro, book, everything but the laptop...
The main thing, i know, is probably getting the keyboard working *cough* but it would be so awesome if it worked because of faster xcode, faster than vmware, etc!

**Comment by Sartharon** [comment]
First of all MacOs is Unix based not linux. Which are different things after all altough Linus Torvalds based some of his code on Unix it's a completey different thing. And the thing/big problem are the drivers in for MacOs which might not be available for your device.

Tldr. No, although it's possible to run MacOs on other devices than Apple products its not as easy as swapping out a kernel and you're goos to go.

**Comment by robot198** [comment]
Some people have made kexts that allow people to use surface pro with touchscreen and keyboard working.
*Sigh* Guess not.

**Comment by qzed** [comment]
It might be possible to do this, but it would take _a lot_ of work. You'd have to write drivers for IPTS (touchscreen) and the surface-acpi module (keyboard, battery, ...).

The difference with older Surface Pro models is that the current devices all use IPTS for the touchscreen instead of something I2C based. On Linux at least this is baked into the graphics driver of the intel GPU (and uses OpenCL kernels for line-smoothing and signal processing), so it could be impossible to implement this as kext altogether.

It's likely that you can implement the surface-acpi as kext (if a proper device framework for the required devices is given, I have no clue about MacOS here). Again, this would be a lot of work, but if anyone want's to do this, feel free to ask me questions about the surface-acpi module.

**Comment by Predatux** [comment]
Por si te interesa
https://github.com/bigsadan/surface-pro-4-hackintosh-10.14.3/
